hack/update-codegen.sh

```diff
@@ -244,32 +244,50 @@ gen-defaulters() {
   done
 }
 
+# TODO: fix the workarounds in this function, files that start with "// +build !ignore_autogenerated"
+# are not used by the conversion-gen tool. This bug should get resolved in https://github.com/kubernetes/kubernetes/issues/101567
 gen-conversions() {
   clean pkg/internal/apis 'zz_generated.conversion.go'
   clean pkg/webhook/handlers/testdata/apis 'zz_generated.conversion.go'
   echo "Generating conversion functions..." >&2
+
   # meta api
   "$conversiongen" --go-header-file hack/boilerplate/boilerplate.generatego.txt \
     --input-dirs github.com/jetstack/cert-manager/pkg/internal/apis/meta/v1 \
     -O zz_generated.conversion
+  
+  # Workaround (see: https://github.com/kubernetes/kubernetes/issues/101567)
+  sed -i 's/\+build \!ignore_autogenerated/TEMP_REPLACE_IGNORE_AUTOGENERATED/g' "pkg/internal/apis/meta/v1/zz_generated.conversion.go"
```

Comment:
Can you write more about this workaround. I skimmed through the linked issue and the workaround that I see there is to override the module to a forked version:
```
replace k8s.io/code-generator => github.com/kmodules/code-generator v0.21.1-rc.0.0.20210428003838-7eafae069eb0

replace k8s.io/gengo => github.com/kmodules/gengo v0.0.0-20210428002657-a8850da697c2
```

If this workaround has advantages then please document how it works and perhaps share it in the linked issue for the benefit of others.

Reply:
The main advantage is that we can work with the latest version of these deps (without depending on a forked version).

Reply:
This comment explains what the probelm is: https://github.com/jetstack/cert-manager/pull/4348/files/d93db827d2e33fd69811acfb8740d6f0f7615800#diff-149dfe7bb29d1191dceae3a52915e750e64b7f87257a5fb309c29d3056e2a95dR247-R248

Reply:
This comment should also explain more clearly what this workaround does: https://github.com/jetstack/cert-manager/pull/4348/files#diff-149dfe7bb29d1191dceae3a52915e750e64b7f87257a5fb309c29d3056e2a95dR247-R249